src/GitVersion.Core/VersionCalculation/VariableProvider.cs

```diff
@@ -2,64 +2,80 @@
 using GitVersion.Configuration;
 using GitVersion.Extensions;
 using GitVersion.Helpers;
-using GitVersion.Logging;
 using GitVersion.OutputVariables;
 
 namespace GitVersion.VersionCalculation;
 
 public class VariableProvider : IVariableProvider
 {
     private readonly IEnvironment environment;
-    private readonly ILog log;
 
-    public VariableProvider(IEnvironment environment, ILog log)
-    {
-        this.environment = environment.NotNull();
-        this.log = log.NotNull();
-    }
+    public VariableProvider(IEnvironment environment) => this.environment = environment.NotNull();
 
-    public VersionVariables GetVariablesFor(SemanticVersion semanticVersion, EffectiveConfiguration configuration, bool isCurrentCommitTagged)
+    public VersionVariables GetVariablesFor(
+        SemanticVersion semanticVersion, EffectiveConfiguration configuration, SemanticVersion? currentCommitTaggedVersion)
     {
-        var isContinuousDeploymentMode = configuration.VersioningMode == VersioningMode.ContinuousDeployment && !isCurrentCommitTagged;
+        var preReleaseTagName = semanticVersion.PreReleaseTag.Name;
+
+        var isContinuousDeploymentMode = configuration.VersioningMode == VersioningMode.ContinuousDeployment
+            && currentCommitTaggedVersion is null;
         if (isContinuousDeploymentMode)
         {
             semanticVersion = new SemanticVersion(semanticVersion);
             // Continuous Deployment always requires a pre-release tag unless the commit is tagged
             if (!semanticVersion.PreReleaseTag.HasTag())
             {
-                var label = configuration.GetBranchSpecificLabel(semanticVersion.BuildMetaData.Branch, null);
-                semanticVersion.PreReleaseTag.Name = label ?? string.Empty;
-                if (semanticVersion.PreReleaseTag.Name.IsNullOrEmpty())
+                // TODO: Why do we manipulating the semantic version here in the VariableProvider? The method name is GET not MANIPULATE.
```

Comment:
I wonder if these TODOs are still relevant taking in consideration we compute the semantic version and return it. We could rename the method to describe the 'Compute' logic

Reply:
Okay no problem I have removed the comment.